Project 5 For John Beverly

```diff
@@ -51,3 +51,441 @@ You may find the following resources useful:
 
 I suggest setting up 15 minute 'stand-up' meetings every other day, where team members are expected to join. zoom call or meet in person and (a) explain what progress they have made on their sub-task, (b) explain what they intend to achieve before the next stand-up, and (c) share any blockers that have to progress. This is an effective way to keep members focused on a team goal, as well as opportunities to overcome challenges. 
+
+
+
+
+Group 2 contains Emanuel, Katie, Josh V (commonly know as V by group member), Cameron, Josh B, Jieming, Lance, Tim(?), also Delaney 
+Please note we DID check this against SHACL playground :) 
+enjoy, if possible.
+
+1.	Constraint for Asymmetry 
+
+a.	Our constraint checks for asymmetric patterns in a graph.
+
+b.	An example of when to use this constraint would be for tracking enrollment in courses at a university. Such that each node would represent a student of course and each relation represents the enrollment of a student in a course. Enrollment relations should only be allowed between students and courses that are in a department. If a violation of this constraint is detected it would mean that there is an enrollment relation that does not satisfy the requirement of a student enrolling in a course without the course being enrolled in a student. By changing ex:AsymmteryShape to ex:AsymmetryEnrollementShape and following our basic structure. 
+
+c.	(a -> b) ^ ~(b -> a)
+
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix ex: <http://example.com/> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+
+ex:AsymmetryShape
+
+a sh:NodeShape ;
+
+sh:targetClass ex:relation ;
+
+sh:property [    
+
+sh:path rdf:value ;    
+
+sh:minCount 1 ;    
+
+sh:maxCount 2 ;    
+
+sh:datatype xsd:string ; 
+
+] .
+
+2.	Constraint for Symmetry 
+a.	Our constraint checks for symmetry patterns in a graph.
+
+b.	An example of when to use this constraint would be if we had an RDF graph that represented your 
+connections on LinkedIn. Such that if A is connected with B then B is connected with A. Wanting to ensure this relation is symmetric we would use a SHACL constraint like this. Easily substituting SymmteryShape for ConnectionSymmteryShape and then following our model below. Our model would alert the user if there were cases of connection not going both ways.
+
+c.	a -> b ^ b -> a
+
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix ex: <http://example.com/> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+ex:SymmetryShape
+
+a sh:NodeShape ;
+
+sh:targetClass ex:relation ;
+
+sh:property [    
+
+sh:path rdf:value ;    
+
+sh:minCount 1 ;    
+
+sh:maxCount 4 ;    
```

Comment:
Hi Karl to your point above about exactly one. That was for an old constraint that we got rid of :) hope that helps!
We chose this value system so that there would be enough instances to check for symmetry. Could you explain further the has_car_seat concern? the SHACL constraints are meant to check that graphs are adhering to specific shapes and I am not sure how one would build a graph for has_car_seat (Although I do not doubt you could be able to build one).

Reply:
I guess I am reviewing the newest version up to yesterday? Here is what I reviewed:

ex:SymmetryShape

a sh:NodeShape ;

sh:targetClass ex:relation ;

sh:property [    

sh:path rdf:value ;    

sh:minCount 1 ;    

sh:maxCount 4 ;    

sh:datatype xsd:string 

] .

You know, this SHACL shape basically says: _Oh, this shape applies to ex:relation, and each instance of ex:relation must have several string values, that is all._ So my general concern still concern: This shape is so general that it is hard to capture some logical property. **Has_car_seat** may satisfy your shape (given all cars have 4 seats at most), but it is clearly not symmtric. So this shape cannot distinguish symmetric relations from non-symmetric relations.

Reply:
I see. Okay I will try to adjust